prow/cmd/deck/static/script.js

```diff
@@ -61,12 +61,229 @@ function redrawOptions(opts) {
+
+    var selectedJobs = activeSearch.getElementsByClassName("job-selected");
+    if (selectedJobs.length > 1) return;
+    if (selectedJobs.length === 0) {
+        // If no job selected, selecte the first one that visible in the list.
+        var activeSearchRect = activeSearch.getBoundingClientRect();
+        var jobs = Array.from(activeSearch.children)
+                        .filter(function (elChild) {
+                            var childRect = elChild.getBoundingClientRect();
+                            var listRect = activeSearch.getBoundingClientRect();
+                            return childRect.top >= listRect.top &&
+                                (childRect.top < listRect.top + listRect.height);
+                        });
+        if (jobs.length === 0) return;
+        jobs[0].classList.add("job-selected");
+        return;
+    }
+    var selectedJob = selectedJobs[0];
+    var nextSibling = selectedJob.nextElementSibling;
+    if (!nextSibling) return;
+
+    selectedJob.classList.remove("job-selected");
+    nextSibling.classList.add("job-selected");
+    adjustScroll(nextSibling);
+}
+
+function handleUpKey() {
+    var activeSearches =
+        document.getElementsByClassName("active-fuzzy-search");
+    if (activeSearches && activeSearches.length !== 1) return;
+    var activeSearch = activeSearches[0];
+    if (activeSearch.tagName !== "UL" ||
+        activeSearch.childElementCount === 0) return;
+
+    var selectedJobs = activeSearch.getElementsByClassName("job-selected");
+    if (selectedJobs.length !== 1) return;
+
+    var selectedJob = selectedJobs[0];
+    var previousSibling = selectedJob.previousElementSibling;
+    if (!previousSibling) return;
+
+    selectedJob.classList.remove("job-selected");
+    previousSibling.classList.add("job-selected");
+    adjustScroll(previousSibling);
+}
+
 window.onload = function() {
+    document.addEventListener("keydown", function(event) {
+        if (event.keyCode === 40) {
+            handleDownKey();
+        } else if (event.keyCode === 38) {
+            handleUpKey();
+        }
+    });
     // set dropdown based on options from query string
-    redrawOptions(optionsForRepo(""));
+	var opts = optionsForRepo("");
+    initFuzzySearch(
+        "job",
+        "job-input",
+        "job-list",
+        Object.keys(opts["jobs"]).sort());
+    redrawOptions(opts);
     redraw();
 };
 
+function displayFuzzySearchResult(el, inputContainer) {
+    el.classList.add("active-fuzzy-search");
+    el.style.top = inputContainer.height + "px";
+    el.style.width = inputContainer.width + "px";
+    el.style.height = 200 + "px";
+}
+
+function fuzzySearch(id, inputId, listId, data, inputValue) {
+    var result = [];
+    inputValue = inputValue.replace("/^\s+|\s+$/g", "");
+
+    if (inputValue === "") {
+        addOptionFuzzySearch(data, id, inputId, listId, true);
+        return;
+    }
+
+    var reg = new RegExp(".*" + inputValue + ".*");
```

Comment:
Similarly here instead of regular expressions, using https://www.w3schools.com/jsref/jsref_includes.asp ?

Reply:
done